identity-settings-component.js

```diff
@@ -0,0 +1,188 @@
+/*!
+ * Identity Settings UI.
+ *
+ * Copyright (c) 2012-2016 Digital Bazaar, Inc. All rights reserved.
+ *
+ * @author Dave Longley
+ * @author Alex Lamar
+ */
+define([
+  'angular',
+  'forge',
+  'lodash'], function(angular, forge, _) {
+
+'use strict';
+
+function register(module) {
+  module.component('brIdentitySettings', {
+    bindings: {
+      identity: '<brIdentity'
+    },
+    controller: Ctrl,
+    templateUrl: requirejs.toUrl(
+      'bedrock-angular-identity/identity-settings-component.html')
+  });
+}
+
+/* @ngInject */
+function Ctrl($scope, config, brAlertService, brIdentityService) {
+  var self = this;
+  self.state = brIdentityService.state;
+  self.help = {};
+  self.emailHash = null;
+  self.imagePreview = null;
+  self.publicMode = 'most';
+  self.public = {};
+  self.loading = true;
+
+  var _gravatarUrl = function() {
+    var url = 'https://secure.gravatar.com/avatar/' + self.emailHash;
+    // use G rating
+    url += '?r=g';
+    if(self.identity.sysGravatarType === 'gravatar') {
+      // default to mystery man
+      url += '&d=mm';
+    } else {
+      // force a custom type
+      url += '&f=y&d=' + self.identity.sysGravatarType;
+    }
+    return url;
+  };
+
+  var _updateImagePreview = function() {
+    switch(self.identity.sysImageType) {
+      case 'url':
+        self.imagePreview = self.identity.image;
+        break;
+      case 'gravatar':
+        // generate gravatar image
+        self.imagePreview = _gravatarUrl();
+        // use 80px x 80px for preview
+        self.imagePreview += '&s=80';
+        break;
+    }
+  };
+  // only update every ~1s, to avoid too many requests while typing
+  var _updateImagePreviewSlowly = _.debounce(function() {
+    _updateImagePreview();
+    $scope.$apply();
+  }, 1000);
```

Comment:
This can be done more efficiently through `br-input-ng-change="$ctrl.foo()"` and `br-input-ng-model-options="{debounce: 1000}"` on the appropriate `br-input` element instead of reinventing it and also using a watch. At a minimum add an issue to fix this up.